File: common/src/main/java/com/sping/common/entity/AfterSale.java
package com.sping.common.entity;

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "after_sale")
public class AfterSale {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "after_sale_id")
    private Integer afterSaleId;

    @Column(name = "order_id", nullable = false)
    private Integer orderId;

    @Column(name = "order_no", nullable = false, length = 32)
    private String orderNo;

    @Column(name = "user_id", nullable = false)
    private Integer userId;

    @Column(name = "user_name", nullable = false, length = 20)
    private String userName;

    @Column(name = "type", nullable = false, length = 20)
    private String type;

    @Column(name = "reason", length = 255)
    private String reason;

    @Column(name = "refund_amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal refundAmount;

    @Column(name = "status", nullable = false, length = 20)
    private String status;

    @Column(name = "admin_remark", length = 255)
    private String adminRemark;

    @Column(name = "create_time", nullable = false)
    private LocalDateTime createTime;

    @Column(name = "review_time")
    private LocalDateTime reviewTime;

    @Column(name = "return_time")
    private LocalDateTime returnTime;

    @Column(name = "refund_time")
    private LocalDateTime refundTime;

    @Column(name = "cancel_time")
    private LocalDateTime cancelTime;

    @Transient
    private String statusText;

    @Transient
    private String typeText;

    public AfterSale() {}

    @PrePersist
    protected void onCreate() {
        this.createTime = LocalDateTime.now();
        if (this.status == null) {
            this.status = "PENDING_REVIEW";
        }
    }

    public Integer getAfterSaleId() { return afterSaleId; }
    public void setAfterSaleId(Integer afterSaleId) { this.afterSaleId = afterSaleId; }
    public Integer getOrderId() { return orderId; }
    public void setOrderId(Integer orderId) { this.orderId = orderId; }
    public String getOrderNo() { return orderNo; }
    public void setOrderNo(String orderNo) { this.orderNo = orderNo; }
    public Integer getUserId() { return userId; }
    public void setUserId(Integer userId) { this.userId = userId; }
    public String getUserName() { return userName; }
    public void setUserName(String userName) { this.userName = userName; }
    public String getType() { return type; }
    public void setType(String type) { this.type = type; }
    public String getReason() { return reason; }
    public void setReason(String reason) { this.reason = reason; }
    public BigDecimal getRefundAmount() { return refundAmount; }
    public void setRefundAmount(BigDecimal refundAmount) { this.refundAmount = refundAmount; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public String getAdminRemark() { return adminRemark; }
    public void setAdminRemark(String adminRemark) { this.adminRemark = adminRemark; }
    public LocalDateTime getCreateTime() { return createTime; }
    public void setCreateTime(LocalDateTime createTime) { this.createTime = createTime; }
    public LocalDateTime getReviewTime() { return reviewTime; }
    public void setReviewTime(LocalDateTime reviewTime) { this.reviewTime = reviewTime; }
    public LocalDateTime getReturnTime() { return returnTime; }
    public void setReturnTime(LocalDateTime returnTime) { this.returnTime = returnTime; }
    public LocalDateTime getRefundTime() { return refundTime; }
    public void setRefundTime(LocalDateTime refundTime) { this.refundTime = refundTime; }
    public LocalDateTime getCancelTime() { return cancelTime; }
    public void setCancelTime(LocalDateTime cancelTime) { this.cancelTime = cancelTime; }
    public String getStatusText() { return statusText; }
    public void setStatusText(String statusText) { this.statusText = statusText; }
    public String getTypeText() { return typeText; }
    public void setTypeText(String typeText) { this.typeText = typeText; }
}

File: sping-frontend/src/api/afterSale.ts
import api from './index'

export interface AfterSaleOrder {
  afterSaleId: number
  orderId: number
  orderNo: string
  userId?: number
  userName?: string
  type: string
  typeText: string
  reason?: string
  refundAmount: number
  status: string
  statusText: string
  adminRemark?: string
  createTime?: string
  reviewTime?: string
  returnTime?: string
  refundTime?: string
  cancelTime?: string
}

export interface ApplyAfterSaleRequest {
  orderId: number
  type: string
  reason?: string
  refundAmount: number
}

export const afterSaleApi = {
  apply(data: ApplyAfterSaleRequest) {
    return api.post('/api/after-sale/apply', data)
  },
  getMyAfterSales() {
    return api.get('/api/after-sale/my')
  },
  getDetail(afterSaleId: number) {
    return api.get(`/api/after-sale/${afterSaleId}`)
  },
  cancel(afterSaleId: number) {
    return api.put(`/api/after-sale/${afterSaleId}/cancel`)
  },
  getAllAfterSales() {
    return api.get('/api/after-sale/admin/list')
  },
  review(afterSaleId: number, action: string, remark?: string) {
    return api.put(`/api/after-sale/admin/${afterSaleId}/review`, { action, remark })
  },
  confirmReturn(afterSaleId: number) {
    return api.put(`/api/after-sale/${afterSaleId}/return`)
  },
  refund(afterSaleId: number) {
    return api.put(`/api/after-sale/admin/${afterSaleId}/refund`)
  },
}
